### PR TITLE
feat enable all to data_source of metrics summarize

### LINKED ISSUE #1446: enh(skore): Enable `data_source="all"` in metrics accessor
### Is your feature request related to a problem? Please describe.

As a data scientist, I would like to see if my model is overfitting or not, by comparing the score of my model on the train set and on the test set.

As of v0.9.1, to get the summary of metrics on the test set, I can do:
```python
from sklearn.datasets import fetch_california_housing
from sklearn.model_selection import train_test_split
from sklearn.linear_model import Ridge
from sklearn.pipeline import make_pipeline
from sklearn.preprocessing import StandardScaler
from skore import EstimatorReport

X, y = fetch_california_housing(return_X_y=True, as_frame=True)

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=0)

ridge_report = EstimatorReport(
    make_pipeline(StandardScaler(), Ridge()),
    X_train=X_train,
    X_test=X_test,
    y_train=y_train,
    y_test=y_test,
)

ridge_report.metrics.summarize().frame()  # default on test
```
and for the train:
```python
ridge_report.metrics.summarize(data_source="train").frame() 
```
But I can't have both the train and test sets on the same dataframe.

### Describe the solution you'd like

Have something like:
```python
ridge_report.metrics.summarize(data_source="all").frame()
```
for both train and test sets, which would return this kind of dataframe:

Image

### Describe alternatives you've considered, if relevant

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by MatthieuLoustaunau**
Closes #1446

### CODE REVIEW

**Comment by MatthieuLoustaunau** [comment]
missing : 
- [x] Favorability appears twice with data_source="all"
- [x] doc to update.

**Comment by MarieSacksick** [comment]
@MatthieuLoustaunau, do you want to finish this PR, or should we take it from here?

**Comment by MatthieuLoustaunau** [comment]
I plan to come back at this during the next week, as I will have some spare time. If I cannot finish within 2 weeks I let you take over.

**Comment by MatthieuLoustaunau** [comment]
@MarieSacksick  I have removed the `_summarize` method, hopefully avoiding confusion with `summarize`.

File: skore/src/skore/_sklearn/_estimator/metrics_accessor.py [hunk 1]
```diff
@@ -429,6 +489,18 @@ class is set to the one provided when creating the report. If `None`,
...
+            X=X,
+            y=y,
+            scoring=scoring,
+            scoring_names=scoring_names,
+            scoring_kwargs=scoring_kwargs,
+            pos_label=pos_label,
+            indicator_favorability=indicator_favorability,
+            flat_index=flat_index,
+        )
+
```
Comment (MarieSacksick) [inline skore/src/skore/_sklearn/_estimator/metrics_accessor.py]: It looks like this is no useful anymore

File: skore/src/skore/_sklearn/_estimator/metrics_accessor.py [hunk 2]
```diff
@@ -77,12 +77,13 @@ def summarize(
...
         Parameters
         ----------
-        data_source : {"test", "train", "X_y"}, default="test"
+        data_source : {"test", "train", "X_y", "all"}, default="test"
             The data source to use.
 
             - "test" : use the test set provided when creating the report.
             - "train" : use the train set provided when creating the report.
             - "X_y" : use the provided `X` and `y` to compute the metric.
+            - "all" : use both the train and test sets to compute the metrics.
```
Comment (MarieSacksick) [inline skore/src/skore/_sklearn/_estimator/metrics_accessor.py]:
```suggestion
            - "all" : use both the train and test sets to compute the metrics and present them side by side
```

To make it more explicit that it's not doing an average or anything of that kind

File: skore/src/skore/_sklearn/_estimator/metrics_accessor.py [hunk 3]
```diff
@@ -63,7 +63,7 @@ def __init__(self, parent: EstimatorReport) -> None:
     def summarize(
         self,
         *,
-        data_source: DataSource = "test",
+        data_source: Literal["test", "train", "X_y", "all"] = "test",
```
Comment (MarieSacksick) [inline skore/src/skore/_sklearn/_estimator/metrics_accessor.py]: Instead of redefining completely DataSource, could you extend if with "all" please? (with `| "all"`, or by defining `from typing import Union, SummarizeDataSource = Union[DataSource, Literal["all"]]` for instance)

File: skore/src/skore/_sklearn/_estimator/metrics_accessor.py [hunk 4]
```diff
@@ -160,7 +161,66 @@ class is set to the one provided when creating the report. If `None`,
...
+                    test_df,
+                ],
+                axis=1,
+            )
+            if indicator_favorability:
+                combined["Favorability"] = train_df["Favorability (train)"]
+                combined.drop(
+                    columns=["Favorability (train)", "Favorability (test)"],
+                    inplace=True,
+                )
```
Comment (MarieSacksick) [inline skore/src/skore/_sklearn/_estimator/metrics_accessor.py]: Nitpick: instead, you could by default set favorability to false for test set for instance, and drop only the suffix. It will avoid some table manipulation.

File: skore/src/skore/_sklearn/_estimator/metrics_accessor.py [hunk 5]
```diff
@@ -160,7 +161,66 @@ class is set to the one provided when creating the report. If `None`,
...
+        ... ).frame()
+                                  LogisticRegression Favorability
+        Metric   Label / Average
+        F1 Score               1             0.96...          (↗︎)
         """
+        if data_source == "all":
+            train_summary = self.summarize(
+                data_source="train",
+                X=X,
+                y=y,
```
Comment (MarieSacksick) [inline skore/src/skore/_sklearn/_estimator/metrics_accessor.py]: you don't need to pass X and y here, since it's the data_source that you want to use (same below for test)
